### PR TITLE
[JUJU-1667] Remove add-subnet command

### PR DESCRIPTION
**Pull request by manadart**
This command has a genesis in the spaces MVP from years ago. Today it is functionally void.
- Attempting to add a subnet that does not exist results in a `not found` error for the CIDR.
- Attempting to add a subnet where the CIDR exists results in an `already exists` error.
- The command cannot be used to move a subnet between spaces.

Here, we remove the command and associated client logic. The subnets client facade is bumped to a new version 5, without the `AddSubnets` method.

This will be followed by a patch to the develop branch that removes the server-side logic entirely.

## QA steps

Check that `add-subnet` is not available.

## Documentation changes

We will have to remove references from documentation.

## Bug reference

N/A

### CODE REVIEW

File: apiserver/facades/client/subnets/subnets.go [hunk 1]
```diff
@@ -144,12 +149,8 @@ func (api *APIv3) AllSpaces() (params.SpaceResults, error) {
 	return results, nil
 }
 
-// AllSpaces does not belong on this API facade.
-// If you need it, use the spaces facade.
-func (api *API) AllSpaces(_, _ struct{}) {}
```
Comment (manadart) [inline apiserver/facades/client/subnets/subnets.go:149]: We didn't actually need to be blanking this on the inner-most API.

**Review by hmlanigan — COMMENTED** [review]
The code LGTM.

I'm curious when the command became functionally void? Should we leave it in the juju 2.9 client for backwards compatibility only?

**Comment by manadart** [comment]
I don't know if it was ever useful. If it was, my guess is that the space/subnet _backing_ would have included subnets detected from the provider. This was written prior to `reload-spaces`.

Back-porting to 2.9 was suggested by @wallyworld, as bumping the facade version there allows complete removal of server-side logic from 3.0. This was also discussed with @jameinel who concurred, despite the reservations of removal in a point release.

**Review by hmlanigan — APPROVED** [review]

**Comment by barrettj12** [comment]
@manadart can you merge this through to develop as well? There are conflicts in the facade which I'm not sure how to resolve.

**Comment by manadart** [comment]
> @manadart can you merge this through to develop as well? There are conflicts in the facade which I'm not sure how to resolve.

https://github.com/juju/juju/pull/14599

**Pull request merged**